pkg/operation/botanist/kubeapiserver.go

```diff
@@ -122,7 +122,17 @@ func (b *Botanist) computeKubeAPIServerAutoscalingConfig() kubeapiserver.Autosca
 	if hvpaEnabled {
 		nodeCount = b.Shoot.GetMaxNodeCount()
 	}
-	apiServerResources = resourcesRequirementsForKubeAPIServer(nodeCount, b.Shoot.GetInfo().Annotations[v1beta1constants.ShootAlphaScalingAPIServerClass])
+
+	if features.DefaultFeatureGate.Enabled(features.DisableScalingClassesForShoots) {
```

Comment:
As per the release notes, resource requests for `kube-apiserver`s are assigned statically only if
- Feature gate `DisableScalingClassesForShoots` is turned on
-  `HVPA` feature gate is enabled for Seed.
```suggestion
	if hvpaEnabled && features.DefaultFeatureGate.Enabled(features.DisableScalingClassesForShoots) {
```

Reply:
That's a great catch, thanks! I was under the assumption that the case where HVPA disabled was sufficiently covered by https://github.com/gardener/gardener/blob/84f14c71516aeabe07e664a61f47a20051519a2f/pkg/operation/botanist/kubeapiserver.go#L145-L153 but it turns out, this is only the case for a ManagedSeed. For regular Shoot, we still need the check just like you suggested.

To make this visible, I 
* changed the tests to actually look at the Node count and verify if `maxNodeCount` or `minNodeCount` is taken into account for computing resource requests
* Added tests for the HVPA disabled case for Shoots and ManagedSeeds
* moved the definition of the `hvpaEnabled` boolean up, such that also for ManagedSeeds the "normal" case is that their `maxNodeCount` is taken into account for computing the `ScalingClass` – just like for Shoot clusters

Reply:
For the shoot, changes look good. 
Here are a few points for Managed Seed API server resource requests - 
- Old scenario - 

  - Previously for managed seed if feature gate `HVPAForShootedSeed` was disabled, API server resource request used to get assigned statically ref - https://github.com/gardener/gardener/blob/a7ff44596dfe18b423d94df589f5d9dbd93433a1/pkg/operation/botanist/kubeapiserver.go#L135-L142
  - And if feature gate `HVPAForShootedSeed` is enabled,  API server resource request used to get assigned based on `ScalingClass` and node count was decided based on `HVPA` feature gate ref - https://github.com/gardener/gardener/blob/a7ff44596dfe18b423d94df589f5d9dbd93433a1/pkg/operation/botanist/kubeapiserver.go#L121-L125

- New Scenario (This PR)- 
  - If feature gate `HVPAForShootedSeed` is disabled API server resource request will get assigned the same as the old scenario.
  - If `HVPAForShootedSeed` is enabled and feature gate `DisableScalingClassesForShoots` is disabled resource will get assigned based on `ScalingClass` and node count will get calculated based on `HVPAForShootedSeed` feature gate and not based on `HVPA` feature gate(Old case). I am not sure about this, if we should decide node count based on `HVPAForShootedSeed` or should follow the old way and decide node count based on `HVPA` feature gate.
  - If `HVPAForShootedSeed` is enabled and feature gate `DisableScalingClassesForShoots` enabled, requests resource will get assigned statically but values are 
    ```
      corev1.ResourceCPU:    resource.MustParse("500m"),
      corev1.ResourceMemory: resource.MustParse("1Gi"),
    ```
     Here Should we not assign higher resource requests same as the old case values.
    ```
     corev1.ResourceCPU:    resource.MustParse("1750m"),
     corev1.ResourceMemory: resource.MustParse("2Gi"),
    ```
    Or will it be fine to start with `500m` CPU and `1Gi` memory.

Reply:
> Or will it be fine to start with 500m CPU and 1Gi memory.

IMO, this should be fine for now. We should avoid any new special handling which complicates the code. Newly created managed seeds can just scale up their API servers like for regular shoots.

Reply:
@voelzmo what about this point -
`
If HVPAForShootedSeed is enabled and feature gate DisableScalingClassesForShoots is disabled resource will get assigned based on ScalingClass and node count will get calculated based on HVPAForShootedSeed feature gate and not based on HVPA feature gate(Old case). I am not sure about this, if we should decide node count based on HVPAForShootedSeed or should follow the old way and decide node count based on HVPA feature gate.
`
Do you want to keep this change?

Reply:
I think with Rafael's comment above we agreed to keep it like it is in this change – or did I misunderstand that?

Reply:
I think Rafael's comment was about the amount of memory and CPU we are assigning.

Reply:
/cc @rfranzke

Reply:
Ok, at this point, I'm not sure what to do with this comment thread. I propose to leave the code as-is and get it merged. Any objections?

Reply:
Ok, from my side.